apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/AssessmentQuestionManualGrading.tsx

```diff
@@ -0,0 +1,205 @@
+import type { RubricData } from '../../../lib/manualGrading.types.js';
+
+import { GroupInfoModal } from './GroupInfoModal.js';
+import type { InstanceQuestionRowWithAIGradingStats as InstanceQuestionRow } from './assessmentQuestion.types.js';
+import { AssessmentQuestionTable } from './assessmentQuestionTable.js';
+
+const queryClient = new QueryClient();
+
+export interface AssessmentQuestionManualGradingProps {
+  authzData: PageContextWithAuthzData['authz_data'];
+  course: StaffCourseInstanceContext['course'];
+  courseInstance: StaffCourseInstanceContext['course_instance'];
+  csrfToken: string;
+  instanceQuestions: InstanceQuestionRow[];
+  urlPrefix: string;
+  assessmentId: string;
+  assessmentQuestionId: string;
+  assessmentQuestion: AssessmentQuestion;
+  assessmentTid: string;
+  questionQid: string;
+  aiGradingMode: boolean;
+  groupWork: boolean;
+  rubricData: RubricData | null;
+  instanceQuestionGroups: InstanceQuestionGroup[];
+  courseStaff: { user_id: string; name: string | null; uid: string }[];
+  aiGradingStats: AiGradingGeneralStats | null;
+  numOpenInstances: number;
+  search: string;
+  isDevMode: boolean;
+}
+
+type AssessmentQuestionManualGradingInnerProps = Omit<
+  AssessmentQuestionManualGradingProps,
+  'search' | 'isDevMode'
+>;
+
+function AssessmentQuestionManualGradingInner({
+  authzData,
+  instanceQuestions,
+  course,
+  courseInstance,
+  urlPrefix,
+  csrfToken,
+  assessmentId,
+  assessmentQuestion,
+  assessmentTid,
+  questionQid,
+  aiGradingMode,
+  groupWork,
+  rubricData,
+  instanceQuestionGroups,
+  courseStaff,
+  aiGradingStats,
+  numOpenInstances,
+}: AssessmentQuestionManualGradingInnerProps) {
+  const [showSelectedModal, setShowSelectedModal] = useState(false);
+  const [showAllModal, setShowAllModal] = useState(false);
+  const [showUngroupedModal, setShowUngroupedModal] = useState(false);
+  const [selectedIdsForGrouping, setSelectedIdsForGrouping] = useState<string[]>([]);
+
+  const handleShowSelectedModal = (ids: string[]) => {
+    setSelectedIdsForGrouping(ids);
+    setShowSelectedModal(true);
+  };
+
+  const groupSubmissionMutation = useMutation<
+    { jobSequenceId?: string; success: boolean },
+    Error,
+    { action: string; closedOnly: boolean; instanceQuestionIds?: string[] }
+  >({
+    mutationFn: async ({ action, closedOnly, instanceQuestionIds }) => {
+      const requestBody: Record<string, any> = {
+        __csrf_token: csrfToken,
+        __action: action,
+      };
+
+      if (action === 'batch_action') {
+        requestBody.batch_action = 'ai_instance_question_group_selected';
+        requestBody.instance_question_id = instanceQuestionIds || [];
+      }
+
+      if (numOpenInstances > 0) {
+        requestBody.closed_instance_questions_only = closedOnly;
+      } else {
+        requestBody.closed_instance_questions_only = false;
+      }
+
+      const response = await fetch('', {
```

Comment:
👀 I'd never seen `fetch('')` before. Does this just mean "fetch to the page's current URL"?